### PR TITLE
[Libbeat] Respect the `parameters` option defined in the ES output.

### PR DESCRIPTION
**Pull request by ph**
This make sure that any values defined in the `parameters` option of the
ES output is correctly added to every bulk request query string.

Fixes: #18325

## Checklist

- [x] My code follows the style guidelines of this project
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have made corresponding change to the default configuration files
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] I have added an entry in `CHANGELOG.next.asciidoc` or `CHANGELOG-developer.next.asciidoc`.

## Author's Checklist

- [ ]

## Related issues

-

### CODE REVIEW

**Comment by ph** [comment]
Adding the _prefer_v2_parameters_ and this PR will allow ES to correctly use the right template.
```
 │ info [o.e.c.m.MetadataCreateIndexService] [winterfell] applying create index request using v2 template [bar]
   │ info [o.e.c.m.MetadataCreateIndexService] [winterfell] [test-ok] creating index, cause [auto(bulk api)], templates [bar], shards [1]/[2], mappings [_doc]
```

**Comment by ph** [comment]
7.8 backport is ready as part of https://github.com/elastic/beats/pull/18318
going to make a 7.x backport after.

File: libbeat/esleg/eslegclient/bulkapi_mock_test.go [hunk 1]
```diff
@@ -146,3 +149,57 @@ func TestOneHost503Resp_Bulk(t *testing.T) {
...
+	client, _ := NewConnection(ConnectionSettings{
+		Parameters: map[string]string{"hello": "world"},
+		URL:        "http://localhost",
+		Timeout:    0,
+	})
+	client.Encoder = NewJSONEncoder(nil, false)
+
+	client.HTTP = &reqInspector{
+		assert: func(req *http.Request) (*http.Response, error) {
+			assert.Equal(t, "world", req.URL.Query().Get("hello"))
```
Comment (urso) [inline libbeat/esleg/eslegclient/bulkapi_mock_test.go:163]: The change modifies the merging of parameters. We should check if all parameters (preconfigured and local ones) are present.

File: libbeat/esleg/eslegclient/bulkapi_mock_test.go [hunk 2]
```diff
@@ -146,3 +149,57 @@ func TestOneHost503Resp_Bulk(t *testing.T) {
...
+
+	client.Bulk(context.Background(), index, "type1", params, body)
+}
+
+type reqInspector struct {
+	assert func(req *http.Request) (*http.Response, error)
+}
+
+func (r *reqInspector) Do(req *http.Request) (*http.Response, error) {
+	return r.assert(req)
```
Comment (urso) [inline libbeat/esleg/eslegclient/bulkapi_mock_test.go:201]: calling `assert` directly has the drawback of not getting triggered if the call to `Bulk` returns early for whatever reason (we do not check the return of `Bulk` to match our expected error). A more flexible and resilient approach would be to 'record' the request into an array and do the validation afterwards.

**Comment by ruflin** [comment]
Backport for 7.8: https://github.com/elastic/beats/pull/18318
Backport for 7.x: https://github.com/elastic/beats/pull/18348

**Pull request merged**